tracer/src/Datadog.Trace/Iast/SensitiveData/EvidenceRedactor.cs

```diff
@@ -97,10 +97,17 @@ public void Process(Source source)
 
     internal Vulnerability RedactVulnerability(Vulnerability vulnerability)
     {
+        if (vulnerability.Evidence?.Value is null)
+        {
+            return vulnerability;
+        }
+
         List<Range>? sensitive = null;
+        string evidenceValue = vulnerability.Evidence?.Value!;
```

Comment:
`Vulnerability.Evidence.Value` is guaranteed not null here. I don't know why you need the `?!` 🤔 

```suggestion
        string evidenceValue = vulnerability.Evidence.Value;
```

Reply:
I get an error. For some reason, it does not detect that the null value has already been checked.

Reply:
That's... weird... 🤔  In this case, I would use `!` where _we_ know it's null (as using `?` actually changes the generated code)

Reply:
Done!